### DISCUSSION TITLE
Token transfer in a smart contract that has custom fees.

### DISCUSSION THREAD

**Discussion by fnLog0** [hashgraph/hedera-smart-contracts#102] [Q&A]
How does Smart Contract Deal With custom fees inside the smart contract?

**Comment by Nana-EC** [comment]
Hi @0xnasim 
Please take a look at [HIP 514](https://hips.hedera.com/hip/hip-514) and it's implementation in [IHederaTokenService](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol) in this repo.

Tokens can be created with the desired set of custom fees using [createFungibleTokenWithCustomFees](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L428) and [createNonFungibleTokenWithCustomFees](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L451)

I hope this helps.

**Reply by fnLog0** [reply]
Hi @Nana-EC 

Basically, I have a problem with transferring tokens inside the smart contract that has custom fees.

Suppose I have a token A that has a custom fee (F). I want to transfer the token to address X using a smart contract.

Have any example of that?
